src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -187,16 +185,15 @@ impl Executor for IndexScanExecutor {
         self.count = 0;
     }
 
-    fn collect_statistics_into(&mut self, statistics: &mut Statistics) {
-        statistics.add(&self.statistics);
-        self.statistics = Statistics::default();
+    fn collect_metrics_into(&mut self, metrics: &mut ExecutorMetrics) {
+        metrics.merge(&mut self.metrics);
         if let Some(scanner) = self.scanner.take() {
-            scanner.collect_statistics_into(statistics);
+            scanner.collect_statistics_into(&mut metrics.cf_stats);
```

Comment:
I think you can adjust here to support `collect multi-times for streaming`:
```rust
if let Some(scanner) = self.scanner.as_mut() {
    // do collect.
}
```

Reply:
Maybe we could fix it in streaming?